src/frequenz/dispatch/_bg_service.py

```diff
@@ -23,6 +28,22 @@
 """The logger for this module."""
 
 
+class MergeStrategy(ABC):
+    """Base class for strategies to merge running intervals."""
+
+    @abstractmethod
+    def filter(self, dispatches: dict[int, Dispatch], dispatch: Dispatch) -> bool:
```

Comment:
I would offer a read-only view over the dispatches, as we don't want anyone to modify the dict!

```suggestion
    def filter(self, dispatches: Mapping[int, Dispatch], dispatch: Dispatch) -> bool:
```